x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptMappedFieldType.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.mapper;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.TextSearchInfo;
+import org.elasticsearch.index.query.QueryShardContext;
+import org.elasticsearch.script.Script;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.search.SearchService.ALLOW_EXPENSIVE_QUERIES;
+
+/**
+ * Abstract base {@linkplain MappedFieldType} for scripted fields.
+ */
+abstract class AbstractScriptMappedFieldType extends MappedFieldType {
+    protected final Script script;
+
+    AbstractScriptMappedFieldType(String name, Script script, Map<String, String> meta) {
+        super(name, false, false, TextSearchInfo.NONE, meta);
+        this.script = script;
+    }
+
+    protected abstract String runtimeType();
+
+    void mapperXContentBody(XContentBuilder builder, Params params) throws IOException {
+        builder.field("runtime_type", runtimeType());
+        builder.field("script", script.getIdOrCode()); // TODO For some reason this doesn't allow us to do the full xcontent of the script.
```

Comment:
I no longer remember what this TODO is about. what can we do to address it? and what do we need this method for, again?

Reply:
It spits out the script in the xcontent of the mapper. I think we might want to merge it into the `FieldMapper`'s xcontent rendering. I can have a look in a follow up.

That TODO is basically complaining that I can render the whole script - it won't parse properly if I do. I don't remember why. I'll dig.

Reply:
I opened #59813 for the TODO.